### PR TITLE
✨ Keycloak

### PR DESCRIPTION
**Pull request by ad2ien**
#305 

(Re)introducing keycloak as an ID provider managed by EIMIS

This time it uses a customize keycloak image : [eimis-ans/eimis-keycloak](https://github.com/eimis-ans/eimis-keycloak)

A realm is configured to with smtp settings and e-mail OTP as 2FA

Tested on lagraulet environnement.

### don't forget before merging

- [x] to modify `SMTP_NOTIFY_FROM`  -> `SMTP_DISPLAY_NAME`, `SMTP_FROM` 
- [x] add  `KEYCLOAK_CLIENT_ID` and `KEYCLOAK_CLIENT_SECRET`

### CODE REVIEW

**Comment by ad2ien** [comment]
#319 to track to next move about keycloak

File: ansible/roles/keycloak-operator/tasks/main.yml
Comment (heroinedor) [inline ansible/roles/keycloak-operator/tasks/main.yml:1]: don't forget to add a switch "present_or_absent" in order to be able to delete the namespace and its content when destroying the environment

File: ansible/roles/keycloak/templates/keycloak.yml.j2 [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    app: keycloak-eimis
+spec:
+  instances: 1
+  image: eimisans/eimis-keycloak:21.1.1
+  hostname:
+    hostname: "{{ keycloak.server_name }}"
+    strict: false
+    strictBackchannel: false
+  ingress:
+    enabled: false
```
Comment (heroinedor) [inline ansible/roles/keycloak/templates/keycloak.yml.j2:16]: why is ingress disabled here ?

File: ansible/roles/synapse-extra-config/templates/cm-extraconfig.yml.j2 [hunk 2]
```diff
@@ -11,8 +11,19 @@ data:
...
-{% if prosante_connect.enabled  %}
     oidc_providers:
+      - idp_id: keycloak
+        idp_name: "EIMIS Connect"
+        issuer: "https://{{ keycloak.server_name }}/realms/eimis-realm"
+        client_id: "{{ keycloak.client_id }}"
+        client_secret: "{{ keycloak.client_secret }}"
+        scopes: ["openid", "profile"]
+        user_mapping_provider:
+          config:
```
Comment (heroinedor) [inline ansible/roles/synapse-extra-config/templates/cm-extraconfig.yml.j2:22]: Is this config correct ?
Shouldn't it be like for prosante connect ?
Something like  :
          config:
            localpart_template: "{% raw %}{{ user.preferred_username }}{% endraw %}"
            display_name_template: "{% raw %}{{ user.given_name }} {{ user.family_name }}{% endraw %}"
            email_template: "{% raw %}{{ user.email }}{% endraw %}"

File: ansible/roles/keycloak/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  retries: 30
+  delay: 5
+  when: absent_or_present=="present"
+
+- name: Keycloak instance
+  kubernetes.core.k8s:
+    state: "{{ absent_or_present }}"
+    template: "keycloak.yml.j2"
+
+- name: Wait for the keycloak instance to be up and running
```
Comment (heroinedor) [inline ansible/roles/keycloak/tasks/main.yml:25]: Is there a better way to detect the readiness of keycloak ?

**Review by heroinedor — CHANGES_REQUESTED** [review]

File: ansible/roles/keycloak/templates/keycloak.yml.j2 (hunk 1 above)
Comment (ad2ien) [reply ansible/roles/keycloak/templates/keycloak.yml.j2:16]: because we manage the ingress ourself  so we don't need the operator default ingress

**Review by heroinedor — APPROVED** [review]

**Pull request merged**